### PR TITLE
hack: pass several Ginkgo parameters to the CLI instead of the test binary

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

For example, --ginkgo.slow-spec-threshold had no effect where it was passed before, probably because the CLI was the one generating the messages that depend on it.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @chendave

### CODE REVIEW

**Comment by dims** [comment]
i looked at it in the morning, will look at the changes you just pushed in a little bit

File: hack/ginkgo-e2e.sh
Comment (aojea) [inline hack/ginkgo-e2e.sh:192]: anf this one and ginkgo.timeout?

File: hack/ginkgo-e2e.sh
Comment (chendave) [inline hack/ginkgo-e2e.sh:188]: How to ensure they are active when not using the Ginkgo CLI while you removing them from here?

**Review by chendave — COMMENTED** [review]
I actually cannot get enough context of this change, which testcase will fail without this change?

File: hack/ginkgo-e2e.sh
Comment (chendave) [reply hack/ginkgo-e2e.sh:188]: `{program}` could also be `dlv exec` or `gdb` I think.

**Comment by chendave** [comment]
/hold
for couple of questions.

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:192]: If we move `timeout`, then `dlv test ./test/e2e` will time out after one hour, the default timeout. Perhaps that is okay?
What we are doing now prevents using using a reasonable timeout in `ginkgo --timeout` because it would just get overwritten by the parameter we pass here - it has to be, or this parameter here wouldn't have had an effect.

Considering all this, I am leaning towards moving *all* Ginkgo parameters to the CLI invocation. It's simply cleaner.

Comment (pohly) [reply hack/ginkgo-e2e.sh:192]: Or be even more intelligent: add them to the CLI when using it, otherwise to the suite parameters.

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:188]: > I actually cannot get enough context of this change, which testcase will fail without this change?

Not exactly failing, but not getting reported as intended: tests are marked as "slow" by the Ginkgo CLI after the default 30s.

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh]: I think that's the cleanest approach. I've pushed version with that and tried it locally with the Ginkgo CLI and delve.

File: hack/ginkgo-e2e.sh
Comment (aojea) [inline hack/ginkgo-e2e.sh:202]: can we have compatibility issues with this? I see that we require bash so it should be fine, right?

**Comment by aojea** [comment]
it lgtm but maybe someone with more bash expertise should review

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:202]: `${arg#--}` to strip the prefix of `arg` is even supported by POSIX. Should be fine.

File: hack/ginkgo-e2e.sh
Comment (pohly) [inline hack/ginkgo-e2e.sh:204]: Unsetting `ginkgo_args=()` is no longer necessary - it was in some earlier revision. I can remove this line, or we keep it as reminder that `suite_args` replaces `ginkgo_args`.

Comment (spiffxp) [reply hack/ginkgo-e2e.sh:204]: I would drop this line

File: hack/ginkgo-e2e.sh
Comment (spiffxp) [inline hack/ginkgo-e2e.sh:209]: Setting this unconditionally might be surprising. Consider for followup gating this behind an env var which is enabled if running in CI. Willing to find out if this is too noisy first though.

File: hack/ginkgo-e2e.sh
Comment (spiffxp) [inline hack/ginkgo-e2e.sh:184]: `ginkgo_cli_args` not found in this file, how are tests passing, shouldn't this error out on an unset var?
```suggestion
    program=("${ginkgo}")
```

File: hack/ginkgo-e2e.sh
Comment (spiffxp) [inline hack/ginkgo-e2e.sh:189]: nit
```suggestion
    if [[ -n "${GINKGO_PARALLEL_NODES:-}" ]]; then
      ginkgo_args+=("--nodes=${GINKGO_PARALLEL_NODES}")
    elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
      ginkgo_args+=("--nodes=25")
    fi
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
ginkgo_cli_args needs to be fixed

**Comment by spiffxp** [comment]
In reviewing this I will admit I have forgotten, why do we need to pass things both to ginkgo _and_ the e2e suite?

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:184]: Right, another leftover. Removed.

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:189]: It doesn't make a difference here anymore, but I was trying to reserve `gingko_args` for those arguments that work in both places. I think adding to `program` is more aligned with that.

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:204]: Done

File: hack/ginkgo-e2e.sh
Comment (pohly) [reply hack/ginkgo-e2e.sh:209]: I don't want to change all Prow job definitions just to enabling this particular piece of output. Can we make it depend on some existing env variable?

For example like this:
```
# The following invocation is fairly complex. Let's dump it to simplify
# determining what the final options are. Enabled by default in CI
# environments like Prow.
case "${GINKGO_SHOW_COMMAND:-${CI:-no}}" in y|yes|true) set -x ;; esac
```

**Comment by pohly** [comment]
> In reviewing this I will admit I have forgotten, why do we need to pass things both to ginkgo and the e2e suite?

Not both - either/or.

When using the ginkgo CLI, all Ginkgo parameters can and should be passed to it.

But when using `dlv test ./test/e2e`, then some Ginkgo parameters must be passed to the test suite. `dlv` wouldn't know what to do with them.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold cancel

**Pull request merged**
